### PR TITLE
(fix ca):Updated Ca Certificate Expiration and Rotation by adding a c…

### LINKED ISSUE #743: Parse durations with `y`, `h`, `m` suffixes properly.
ie, when setting up a certificate rotation of `24h` or a CA expiration of `10y`.

Right now we only support `h`.

### PR DESCRIPTION
**Pull request by tharun208**
### Summary

SUMMARY_GOES_HERE
Initially, #750 fixes this issue by using go's inbuilt `ParseDuration` fn. But missed that It won't support for `y`, `m` format. So, this PR contains the fix for that by adding a custom `ParseDuration` function.
### Full changelog

* Added a custom ParseDuration in mesh helper
* Updated tests

### Issues resolved

Fix #743 

Docs in https://github.com/Kong/kuma-website/pull/209

### CODE REVIEW

File: pkg/core/resources/apis/mesh/mesh_helpers.go [hunk 1]
```diff
@@ -70,3 +74,40 @@ func (m *MeshResource) GetCertificateAuthorityBackend(name string) *mesh_proto.C
...
+var durationRE = regexp.MustCompile("^([0-9]+)(y|w|d|h|m|s|ms)$")
+
+// ParseDuration parses a string into a time.Duration
+func ParseDuration(durationStr string) (time.Duration, error) {
+	// Allow 0 without a unit.
+	if durationStr == "0" {
+		return 0, nil
+	}
+	matches := durationRE.FindStringSubmatch(durationStr)
+	if len(matches) != 3 {
```
Comment (nickolaev) [inline pkg/core/resources/apis/mesh/mesh_helpers.go:87]: Will this allow me to do `1y2w3d4h5m6s`?

Comment (tharun208) [reply pkg/core/resources/apis/mesh/mesh_helpers.go:87]: no @nickolaev

Comment (nickolaev) [reply pkg/core/resources/apis/mesh/mesh_helpers.go:87]: Will it make sense to iterate over the matches slice, so such formats can be supported?

Comment (tharun208) [reply pkg/core/resources/apis/mesh/mesh_helpers.go:87]: yes, we can, but thinking that won't be the optimal solution for this.

Comment (nickolaev) [reply pkg/core/resources/apis/mesh/mesh_helpers.go:87]: Yes, I agree. Let's merge this one.

**Pull request merged**